### DISCUSSION TITLE
Skip direct execution

### DISCUSSION THREAD

**Discussion by gowizzard** [go-co-op/gocron#201] [Q&A] [answered]
Hello everyone, I have a question. It may be that I have not found it yet, but I am looking for a way to skip the direct execution at startup. If I define a cronjob that should be executed at 2:00 am and I start the software, then the attached function is executed directly at startup.

Is there a solution for this?

Thanks a lot
Jonas

**Comment by JohnRoesler** [comment, marked as answer]
Hi @jojojojonas glad to have you as a user! 

We do support this functionality both at a job level or for the entire scheduler:

- per job https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.WaitForSchedule
- per scheduler https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.WaitForScheduleAll

**Comment by gowizzard** [comment]
Hallo @JohnRoesler, thank you!

Super this point I have not yet found in the documentation. Thank you very much!
